theseus/geometry/utils/utils.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import threading
+from typing import Any
+
+import torch
+
+
+class LieGroupContext(object):
+    contexts = threading.local()
```

Comment:
Someone said that `threading.local()` is safe for multi-thread processing. Are there any other concerns with this? @luisenp @mhmukadam

Reply:
Nothing comes to mind right now, but we should keep an eye open and tests this thoroughly.

Reply:
Captured in #92.